documentation/projects/proposals/trust_and_safety/content_report_moderation/20230411-project_proposal_content_report_moderation.md

```diff
@@ -0,0 +1,338 @@
+    occurred to me that we could use it to analyse result descriptions during
+    catalog ingestion to evaluate the "usefulness" of a description and develop
+    a score to help increase or reduce descriptions that may not be as useful.
+    This could be expanded to use bigger NLP APIs or text comparison/analysis
+    methods to find other qualities that could factor into the "usefulness" of a
+    description.
+    [I opened this discussion to further explore these ideas](https://github.com/WordPress/openverse/discussions/1175).
+
+Computer vision based enhancements to Django admin are covered in a subsequent
+section as it is sufficiently complicated to warrant its own set of
+considerations.
+
+### Bulk moderation actions (requirement 3)
+
+It is occasionally necessary to apply bulk actions to many results, whether the
+action is deindexing or marking as sensitive. For example, if a spammy Flickr
+user is indexed that only posts ads to pornography websites (before Flickr
+moderates them)[^actual-example], we need to be able to deindex or at least mark
+all the results for that creator as sensitive.
+
+[^actual-example]:
+    This is something that Openverse has actually seen. In other words, this is
+    not an imagined scenario. We need to be able to perform these kinds of bulk
+    actions.
+
+There are two aspects of this that are worth covering in the baseline admin
+improvements:
+
+1. Bulk actions for a specific creator
+1. Bulk actions for a set of results selected via a Django admin list view
+   retrieved based on search terms
+
+To accomplish this, we'll create a new abstract base-model to capture the
+history of bulk moderation actions performed. The base class will require the
+user who performed the action, the date of the action, the type of the action
+(described below), and required notes describing why the action was taken. The
+possible actions are:
+
+- Deindex
+- Mark sensitive
+- Reindex
+- Undo mark sensitive
+
+The last two actions are necessary for "undoing" the first two.
+
+Subclasses must implement a method that returns the list of result identifiers
+to take the action for. Each of the actions will trigger the creation or
+deletion of corresponding `AbstractDeletedMedia` or `AbstractSensitiveMedia`
+(renamed as part of requirement 1 from `AbstractMatureMedia`) for each work
+returned by the subclass method. The subclasses must also implement some way of
+tracking all the works affected by the action.
+
+For the bulk actions list view, the subclass should have a related join table
+tying the bulk action instance identifier to the result identifiers. For the
+creator actions, the subclass only needs to indicate the identifier for the
+creator. Note that disambiguated creator identification must include the
+provider as the same username across different providers may be used by
+different distinct creators.
+
+### Computer vision based moderator safety tools (requirement 4)
```

Comment:
> ### Computer vision based moderator safety tools (requirement 4)

Could this section or the previous about amendments to Django be more explicit that we intend to render the images in the Django UI, and provide text hints? It would probably make sense to decouple the blurring from the computer vision section, actually, because we'd want to blur all reported images by default. We'll need to factor that blurring into the "Content report admin view upgrades" task.

This would help clarify the document, but I think it would also help me as a reviewer to make a determination about whether computer vision is necessary for this project or not. The hints provided by the computer vision API would be supplemental to the mandatory description provider by user reports, but it's already likely that moderators will need to view every reported image. I understand the hints could still improve safety and prepare moderators for what they may see.

Perhaps there would be ways to increase the usefulness of user-reports? For example, beyond a mere text description we could require them to choose content labels in the UI, for things like "violence", "sexual acts", and so on. 

I also wonder about it on the basis of storing the rekognition labels in a way they can be later applied to records in the catalog. That sounds like a potentially complicated and time-consuming dependency to add to this project.

Reply:
Yes, definitely worth clarifying. That was context I had in my head that I forgot to put down in the proposal, thanks for pointing it out.

> mandatory description provider by user reports

User descriptions on sensitive reports are optional, FWIW.

We could also treat the computer vision hints feature as non-blocking for starting to use the process, i.e., as the first improvement to the process that we'll roll out after the "MVP" is finished.

> I also wonder about it on the basis of storing the rekognition labels in a way they can be later applied to records in the catalog. That sounds like a potentially complicated and time-consuming dependency to add to this project.

My idea for this was to throw them in Redis and defer normalisation until the subsequent project, but I should be clear about that. We need to cache them _somehow_ to avoid making duplicate requests (and paying for them) and so long as the key in Redis is normalised the data would still be accessible for the downstream project.

Reply:
> Perhaps there would be ways to increase the usefulness of user-reports? For example, beyond a mere text description we could require them to choose content labels in the UI, for things like "violence", "sexual acts", and so on.

This is a great idea, but would require more design and frontend development than I think was originally assumed for this project, so I'd want to clear it with @panchovm especially to make sure he has availability to help with something like that at this stage.

Reply:
> User descriptions on sensitive reports are optional, FWIW.

You're right, thanks for pointing this out. I was thinking of the "Other" reports mandating a 20-min-character description.